packages/app-project/src/screens/ClassifyPage/ClassifyPageContainer.js

```diff
@@ -10,8 +11,12 @@ function ClassifyPageContainer ({
   workflows = [],
   ...props
 }) {
-  // selected subject state is derived from the page URL, but can be reset by the Classifier component.
+  /* Selected subjectID is initially derived from the page URL
+  via getDefaultPageProps, but subjectID can be reset by the Classifier
+  component via onSubjectReset(), or changed via prioritized subjects UI
+  (Next or Prev buttons) */
   const [selectedSubjectID, setSelectedSubjectID] = useState(subjectID)
```

Comment:
@eatyourgreens I'm confused about this state variable, and hoping you have some insight.

`subjectID` prop is derived from the url and passed from `getStaticProps` to the ClassifyPageContainer  component. Let's say I choose a subject from a subject set on the PRINT project. Then when I use the Next or Prev subject buttons, the url changes due to shallow routing, but `getStaticProps` is not called again.

`selectedSubjectID` is passed to ClassifyPage as `subjectID`, which is used to add the subject to a collection, etc. But how does `selectedSubjectID` stay up to date with shallow routing? From testing production and this PR, I can select various subjects in PRINT and the correct subject always gets added to my collections, but I'm confused as to how that works. I'm missing how `selectedSubjectID` is updated when using the Next or Prev buttons for prioritized subjects.

Reply:
Collections will use the classifier's internal active subject state, which is also copied to the page URL. 

I can't remember what the subject reset does here, but I assume it resets the classifier and clears the current queue.

Reply:
I think this is reset when you cancel subject selection, in order to choose a different subject set.

Reply:
From the blame history, this is the PR that added the subject reset:
https://github.com/zooniverse/front-end-monorepo/pull/2198

Reply:
> Collections will use the classifier's internal active subject state, which is also copied to the page URL.

Oh! I see how that happens via `onAddToCollection` now. I was confused because the variable passed from the classifier's internal active subject is using the same name as ClassiyPage's `subjectID` page. I'll add a clarifying code comment.

I guess my follow-up question then - Is there any reason `selectedSubjectID` should be updated when a volunteer selects a new subject via prioritized subjects UI? The answer doesn't affect bugs this PR fixes, but I might follow-up with a different PR to clarify docs and variable naming.

Reply:
>I guess my follow-up question then - Is there any reason selectedSubjectID should be updated when a volunteer selects a new subject via prioritized subjects UI?

Is `selectedSubjectID` the subject ID that's passed into the classifier to start the subject queue? If so, then selecting a new subject will change that, which will start a new subject queue. Any time that you change the classifier's `workflowID`, `subjectSetID` or `subjectID` props, you start a new subject queue.